### PR TITLE
V2 anonymizer engine readme

### PR DESCRIPTION
**Pull request by shiranr**
Add documentation for new engine.
PLEASE NOTICE:
This commit is on top of 2 others:
https://github.com/microsoft/presidio/pull/535
and:
https://github.com/microsoft/presidio/pull/527
The only thing relevant here is the README.MD file.

### CODE REVIEW

**Review by navalev — APPROVED** [review]

**Comment by omri374** [comment]
Can you please re-open with just the readme file?

File: presidio-anonymizer/README.MD [hunk 1]
```diff
@@ -19,22 +19,20 @@ TODO: change this link to main branch once merged (#2765).
...
-    -   "hash_type" - sets the type of hashing. can be either sha256, sha512 or md5.
-        The default hash type is sha256.
--   FPE - using ff1 algorithm for formatting-Preserving Encryption on the PII
--   Mask - replaces the PII with a given character. <br>
-    Parameters:
-    -   "chars_to_mask" - the amount of characters out of the PII that should
-        be replaced.
-    -   "masking_char" - the character to be replaced with.
-    -   "from_end" - Whether to mask the PII from it's end.
+- Replace - replaces the PII with desired value <br>
```
Comment (omri374) [inline presidio-anonymizer/README.MD:22]: No need for `<br>`. Mkdocs later translates this to HTML and it might pop up there.

File: presidio-anonymizer/README.MD [hunk 2]
```diff
@@ -19,22 +19,20 @@ TODO: change this link to main branch once merged (#2765).
...
+  Parameters: "new_value" - replaces existing text with the given value. <br>
+  If "new_value" is not supplied or empty, default behavior will be: <entity_type>
+  e.g: <PHONE_NUMBER>
+- Redact - removes the PII completely from text Parameters: None
+- Hash - hash the PII using either sha256, sha512 or md5. Parameters:
+    - "hash_type" - sets the type of hashing. can be either sha256, sha512 or md5. The
+      default hash type is sha256.
+- FPE - using ff1 algorithm for formatting-Preserving Encryption on the PII
```
Comment (omri374) [inline presidio-anonymizer/README.MD:30]: Please remove FPE

File: presidio-anonymizer/README.MD [hunk 3]
```diff
@@ -19,22 +19,20 @@ TODO: change this link to main branch once merged (#2765).
...
-        be replaced.
-    -   "masking_char" - the character to be replaced with.
-    -   "from_end" - Whether to mask the PII from it's end.
+- Replace - replaces the PII with desired value <br>
+  Parameters: "new_value" - replaces existing text with the given value. <br>
+  If "new_value" is not supplied or empty, default behavior will be: <entity_type>
+  e.g: <PHONE_NUMBER>
+- Redact - removes the PII completely from text Parameters: None
+- Hash - hash the PII using either sha256, sha512 or md5. Parameters:
+    - "hash_type" - sets the type of hashing. can be either sha256, sha512 or md5. The
```
Comment (omri374) [inline presidio-anonymizer/README.MD:28]: nit: Use \` instead of ": `hash_type` instead of "hash_type"

File: presidio-anonymizer/README.MD [hunk 4]
```diff
@@ -82,7 +80,25 @@ pip install presidio-anonymizer
...
+```python
+from presidio_anonymizer import AnonymizerEngine
+from presidio_anonymizer.entities import AnalyzerResult, AnonymizerConfig
+
+# Initialize the engine with logger.
+engine = AnonymizerEngine()
+
+# Class the anonymize function with the text, analyzer results and
+# Anonymizers config to define the anonymization type.
+result = engine.anonymize(
```
Comment (omri374) [inline presidio-anonymizer/README.MD:92]: Cool :)

File: presidio-anonymizer/README.MD [hunk 5]
```diff
@@ -82,7 +80,25 @@ pip install presidio-anonymizer
...
+    analyzer_results=[AnalyzerResult("PERSON", 11, 15, 0.8),
+                      AnalyzerResult("PERSON", 17, 27, 0.8)],
+    anonymizers_config={"PERSON": AnonymizerConfig("replace", {"new_value": "BIP"})}
+)
+
+print(result)
+
+```
 
 ### As service:
```
Comment (omri374) [inline presidio-anonymizer/README.MD:103]: Not part of this PR, but can you please remove the `pipenv sync` part? Official usage ways are through pip install and Docker, from code is the third option, and it's covered in the installation docs.

File: presidio-anonymizer/README.MD (hunk 1 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:22]: Changed

File: presidio-anonymizer/README.MD (hunk 2 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:30]: Done.

File: presidio-anonymizer/README.MD (hunk 3 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:28]: Done.

File: presidio-anonymizer/README.MD (hunk 5 above)
Comment (shiranr) [reply presidio-anonymizer/README.MD:103]: Done.

Comment (shiranr) [reply presidio-anonymizer/README.MD:103]: Could be this doc is missing all the links to those places???

Comment (shiranr) [reply presidio-anonymizer/README.MD:103]: Lets talk about this tomorrw :) I think there is more to be done.

**Review by omri374 — APPROVED** [review]
